lib/scan/dir/engine.py

```diff
@@ -91,16 +86,28 @@ def check(target, user_agent, timeout_sec, log_in_file, language, time_sleep, th
                                'DESCRIPTION': messages(language, "found").format(target, r.status_code, r.reason),
                                'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd})
             __log_into_file(log_in_file, 'a', data, language)
-            if r.status_code == 200:
-                for dlmsg in directory_listing_msgs:
-                    if dlmsg in content:
-                        info(messages(language, "directoy_listing").format(target))
-                        data = json.dumps({'HOST': target_to_host(target), 'USERNAME': '', 'PASSWORD': '',
-                                           'PORT': "", 'TYPE': 'dir_scan',
-                                           'DESCRIPTION': messages(language, "directoy_listing").format(target), 'TIME': now(),
-                                           'CATEGORY': "scan", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd})
-                        __log_into_file(log_in_file, 'a', data, language)
-                        break
+
+        if r.status_code == 200:
```

Comment:
A seperate **if** because of duplicate matches in the result in case of directories/index found